src/redirect.js

```diff
@@ -0,0 +1,35 @@
+/**
+* If you use a Smartcar hosted redirect this code is served automatically. If
+* you host your own redirect you can source this file (along with sourcing the
+* Javascript SDK on your front end) to close out the redirect and trigger
+* specified `onComplete`
+*/
+
+(function(window) {
+  'use strict';
+  if (!window.opener) {
+    throw new Error('window.opener must be defined');
+  }
+
+  const params = new URLSearchParams(window.location.search);
+
+  const message = {
+    name: 'SmartcarAuthMessage',
+    isSmartcarHosted: window.location.origin === 'https://cdn.smartcar.com',
+    code: params.get('code'),
+    error: params.get('error_description'),
+    state: params.get('state'),
+  };
+
+  // may still host with `app_origin` param even if not using smartcar hosted
```

Comment:
nit: should we make comment style consistent with the comment at the top of the page?

Reply:
Also can we shorten sentences in this paragraph and maybe separate into two chunks (if yes / if no) to make it easier to read?

cc: @charlottekosche